owncloudApp/src/main/java/com/owncloud/android/ui/activity/ManageSpaceActivity.java

```diff
@@ -102,13 +105,14 @@ protected Boolean doInBackground(Void... params) {
             boolean patternEnabled = appPrefs.getBoolean(PatternLockActivity.PREFERENCE_SET_PATTERN, false);
             boolean biometricEnabled = appPrefs.getBoolean(BiometricActivity.PREFERENCE_SET_BIOMETRIC, false);
 
-            String[] passCodeDigits = new String[4];
-            if (passCodeEnable) {
-                passCodeDigits[0] = appPrefs.getString(PassCodeActivity.PREFERENCE_PASSCODE_D1, null);
-                passCodeDigits[1] = appPrefs.getString(PassCodeActivity.PREFERENCE_PASSCODE_D2, null);
-                passCodeDigits[2] = appPrefs.getString(PassCodeActivity.PREFERENCE_PASSCODE_D3, null);
-                passCodeDigits[3] = appPrefs.getString(PassCodeActivity.PREFERENCE_PASSCODE_D4, null);
+
+            final String passcodeString = appPrefs.getString(PREFERENCE_PASSCODE, null);
+            final String[] passCodeDigits = new String[PassCodeActivity.numberOfPassInputs];
+
+            for(int i = 0; i < numberOfPassInputs && passCodeEnable; i++) {
+                passCodeDigits[1] = Character.toString(passcodeString.charAt(i));
```

Comment:
BTW, what does this mean? It assigns a value to the same position on each loop iteration 🤔  Am I missing something?